test/receiveAndDeleteMode.spec.ts

```diff
@@ -0,0 +1,386 @@
+    const receivedMsgs: ServiceBusMessage[] = [];
+    const receiveListener = receiverClient.receive(
+      (msg: ServiceBusMessage) => {
+        receivedMsgs.push(msg);
+        return Promise.resolve();
+      },
+      (err: Error) => {
+        should.not.exist(err);
+      },
+      { autoComplete: autoCompleteFlag }
+    );
+
+    await delay(2000);
+
+    should.equal(receivedMsgs.length, 1);
+    should.equal(receivedMsgs[0].body, testMessages[0].body);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+    should.equal(receivedMsgs[0].body, testMessages[0].body);
+    should.equal(receivedMsgs[0].messageId, testMessages[0].messageId);
+
+    await receiveListener.stop();
+  }
+
+  async function testWithAutoCompleteEnabled(
+    senderClient: QueueClient | TopicClient,
+    receiverClient: QueueClient | SubscriptionClient
+  ): Promise<void> {
+    await sendReceiveMsg(senderClient, receiverClient, true);
+    await testPeekMsgsLength(receiverClient, 0);
+  }
+
+  it("Queue: With auto-complete enabled, no settlement of the message removes message", async function(): Promise<
+    void
+  > {
+    await testWithAutoCompleteEnabled(queueClient, queueClient);
+  });
+
+  it("Subscription: With auto-complete enabled, no settlement of the message removes message", async function(): Promise<
+    void
+  > {
+    await testWithAutoCompleteEnabled(topicClient, subscriptionClient);
+  });
+
+  async function testWithAutoCompleteDisabled(
+    senderClient: QueueClient | TopicClient,
+    receiverClient: QueueClient | SubscriptionClient
+  ): Promise<void> {
+    await sendReceiveMsg(senderClient, receiverClient, false);
+    await testPeekMsgsLength(receiverClient, 0);
+  }
+
+  it("Queue: With auto-complete disabled, no settlement of the message removes message", async function(): Promise<
+    void
+  > {
+    await testWithAutoCompleteDisabled(queueClient, queueClient);
+  });
+
+  it("Subscription: With auto-complete disabled, no settlement of the message removes message", async function(): Promise<
+    void
+  > {
+    await testWithAutoCompleteDisabled(topicClient, subscriptionClient);
+  });
+});
+
+describe("Throws error when Complete/Abandon/Defer/Deadletter/RenewLock of normal message", () => {
+  beforeEach(async () => {
+    await beforeEachTest();
+  });
+
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  async function sendReceiveMsg(
+    senderClient: QueueClient | TopicClient,
+    receiverClient: QueueClient | SubscriptionClient
+  ): Promise<ServiceBusMessage> {
+    await senderClient.send(testMessages[0]);
+    const msgs = await receiverClient.receiveBatch(1);
+
+    should.equal(Array.isArray(msgs), true);
+    should.equal(msgs.length, 1);
+    should.equal(msgs[0].body, testMessages[0].body);
+    should.equal(msgs[0].messageId, testMessages[0].messageId);
+    should.equal(msgs[0].deliveryCount, 0);
+
+    return msgs[0];
+  }
+
+  async function testComplete(
```

Comment:
There isnt much difference between `testComplete()`, `testAbandon()` etc.

Why not rename this to `testSettlement`, make it take another parameter `dispositionType` which would be an enum [DispositionType](https://github.com/Azure/azure-service-bus-node/blob/981c8472b6d94da47607123f3023be5da322e91a/lib/core/messageReceiver.ts#L68). Then you can have switch statement inside to do either of complete()/abandon()/.....